### PR TITLE
fix versal ps kernel host mem mapping issue

### PR DESCRIPTION
**Pull request by mamin506**
#### Problem solved by the commit
VCK5000 is running Linux 5.15. PS kernel cannot create host bo. The reason is Linux DRM API changed.

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
This is found on 5.15 Linux.

#### How problem was solved, alternative solutions (if any) and why they were rejected
Adapter to new Linux DRM API.

#### Risks (if any) associated the changes in the commit
No.

#### What has been tested and how, request additional testing if necessary
VCK5000 PS kernel

#### Documentation impact (if any)
No

### CODE REVIEW

**Review by jeffli-xilinx — APPROVED** [review]
Looks good

File: src/runtime_src/core/edge/drm/zocl/zocl_bo.c [hunk 1]
```diff
@@ -962,6 +962,11 @@ zocl_cma_create(struct drm_device *dev, size_t size)
 	}
 	cma_obj = container_of(gem_obj, struct drm_gem_cma_object, base);
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 11, 0)
+	if (!gem_obj->funcs)
```
Comment (sonals) [inline src/runtime_src/core/edge/drm/zocl/zocl_bo.c:966]: Is this check required?

File: src/runtime_src/core/edge/drm/zocl/include/zocl_drv.h [hunk 2]
```diff
@@ -326,5 +326,6 @@ extern u32 zocl_cu_get_status(struct platform_device *pdev);
 extern u32 zocl_scu_get_status(struct platform_device *pdev);
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 11, 0)
 extern const struct drm_gem_object_funcs zocl_gem_object_funcs;
+extern const struct drm_gem_object_funcs zocl_cma_default_funcs;
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/edge/drm/zocl/include/zocl_drv.h:329]: Why cant we use zocl_gem_object_funcs, Why do we need to create new funcs?

File: src/runtime_src/core/edge/drm/zocl/zocl_bo.c (hunk 1 above)
Comment (mamin506) [reply src/runtime_src/core/edge/drm/zocl/zocl_bo.c:966]: No, we don't really need this check.

**Pull request merged**